discovery_engine_core/web-api2/src/elastic.rs

```diff
@@ -65,18 +76,101 @@ impl Default for Config {
 }
 
 pub(crate) struct ElasticSearchClient {
-    #[allow(dead_code)]
     config: Config,
-    #[allow(dead_code)]
+    url_to_index: Url,
     client: reqwest::Client,
 }
 
 impl ElasticSearchClient {
-    pub(crate) fn new(config: Config) -> Self {
-        Self {
+    pub(crate) fn new(config: Config) -> Result<Self, SetupError> {
+        let mut url_to_index: Url = config.url.parse()?;
+        url_to_index
+            .path_segments_mut()
+            .map_err(|()| "non segmentable url in config")?
+            .push(&config.index_name);
+
+        Ok(Self {
             config,
+            url_to_index,
             client: reqwest::Client::new(),
+        })
+    }
+
+    pub(crate) async fn delete_documents(&self, documents: &[DocumentId]) -> Result<(), Error> {
+        if documents.is_empty() {
+            return Ok(());
         }
+
+        let response = self
+            .bulk_request(
+                documents
+                    .iter()
+                    .map(|document_id| json!({ "delete": { "_id": document_id }})),
+            )
+            .await?;
+
+        if response.errors {
+            let mut errors = Vec::new();
+            for mut response in response.items.into_iter() {
+                if let Some(response) = response.remove("delete") {
+                    if let Ok(status) = StatusCode::from_u16(response.status) {
+                        if status != StatusCode::NOT_FOUND && !status.is_success() {
+                            error!(document_id=%response.id, error=%response.error);
+                            errors.push(response.id);
+                        }
+                    } else {
+                        error!("Non http status code: {}", response.status);
+                    }
```

Comment:
Not sure if we should add `DocumentId` to errors based on status being a valid http status, but rather based on the presence of `error` field in `BulkItemResponse`, wdyt?

Reply:
>the documentation does give guarantees

Let's leave it for now then. We can see later if we can trust their documentation 😉 

>I think we should just use the official rust bindings even if they still are alpha

Let's maybe put a pin in it until Andrea is back. This was an ad hoc decision that was made when the api was a PoC stage, so we can definitely discuss switching to official bindings I think.